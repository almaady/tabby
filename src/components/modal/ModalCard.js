import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

import ButtonIcon from "../buttons/ButtonIcon";
/**
 * @class ModalCard - Is the card for the modals and it contains a cross icon by default
 */
class ModalCard extends React.PureComponent {
  /**
   * @property {func} onClose - The function to close the modal
   * @property {node} children - The content of the modal
   * @property {string} className - Just in case you need another class
   */
  static propTypes = {
    onClose: PropTypes.func.isRequired,
    children: PropTypes.node.isRequired,
    className: PropTypes.string
  };

  render() {
    const { onClose, children, className } = this.props;
    return (
      <div className={classNames("modal", className)}>
        <div className="icons">
          <ButtonIcon icon="cross" color="gray-l2" size="L" onClick={onClose} />
        </div>
        <div className="modal--content">{children}</div>
      </div>
    );
  }
}

ModalCard.defaultProps = {
  className: ""
};
export default ModalCard;
