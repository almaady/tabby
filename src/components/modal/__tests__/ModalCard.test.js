import React from "react";
import { shallow } from "enzyme";
import sinon from "sinon";

import ModalCard from "../ModalCard";

describe("Modal Card", () => {
  const close = sinon.stub();
  let props;
  beforeEach(() => {
    props = {
      onClose: close,
      children: <div />
    };
  });
  it("should render correctly with default state", () => {
    const component = shallow(<ModalCard {...props} />);
    expect(component).toMatchSnapshot();
  });
  it("should call OnClose when you click on button", () => {
    const component = shallow(<ModalCard {...props} />);
    component.find("ButtonIcon").simulate("click");
    expect(close.called).toBeTruthy();
  });
});
