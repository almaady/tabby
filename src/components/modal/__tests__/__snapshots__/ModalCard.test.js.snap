// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Card should render correctly with default state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModalCard
    className=""
    onClose={[Function]}
  >
    <div />
  </ModalCard>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="icons"
        >
          <ButtonIcon
            borderColor=""
            className=""
            color="gray-l2"
            disabled={false}
            icon="cross"
            onClick={[Function]}
            size="L"
            type="button"
          />
        </div>,
        <div
          className="modal--content"
        >
          <div />
        </div>,
      ],
      "className": "modal",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ButtonIcon
            borderColor=""
            className=""
            color="gray-l2"
            disabled={false}
            icon="cross"
            onClick={[Function]}
            size="L"
            type="button"
          />,
          "className": "icons",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "borderColor": "",
            "className": "",
            "color": "gray-l2",
            "disabled": false,
            "icon": "cross",
            "onClick": [Function],
            "size": "L",
            "type": "button",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div />,
          "className": "modal--content",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="icons"
          >
            <ButtonIcon
              borderColor=""
              className=""
              color="gray-l2"
              disabled={false}
              icon="cross"
              onClick={[Function]}
              size="L"
              type="button"
            />
          </div>,
          <div
            className="modal--content"
          >
            <div />
          </div>,
        ],
        "className": "modal",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ButtonIcon
              borderColor=""
              className=""
              color="gray-l2"
              disabled={false}
              icon="cross"
              onClick={[Function]}
              size="L"
              type="button"
            />,
            "className": "icons",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "borderColor": "",
              "className": "",
              "color": "gray-l2",
              "disabled": false,
              "icon": "cross",
              "onClick": [Function],
              "size": "L",
              "type": "button",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div />,
            "className": "modal--content",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
